MetaMask/metamask-extension
# [TS dashboard] Reorganize files slightly

## Explanation

I'm making some usability improvements to the TypeScript dashboard, and this is a prerequisite for those changes that cleans things up a bit.

* Create a better distinction between app files and non-app files by
  moving app files from `src/` to `app/`
* Split up stylesheet file
* Move the partitions file (a file generated only once in watch mode
  that is used to build the visualization on the dashboard) within the
  `build/` directory, as it's a built file
* Extract all file and directory paths to a file shared between the app
  and build scripts
  * Note that we use the data from the partitions file from within the
    app, but because neither `fs.readFile*` nor `path.resolve` works
    within a browser context, we use `brfs` to read the file at build
    time and then bake it into the compiled version of the app. However,
    `brfs` is a bit weird to work with, so we can't completely separate
    the variable that holds the path and the call to `fs.readFile` —
    we have to use them in the same place. See the comment in
    `development/ts-migration-dashboard/common/partitions-file.ts` for
    more.

## Manual Testing Steps

* Run `yarn ts-migration:dashboard:watch`, then run `open development/ts-migration-dashboard/build/final/index.html`. You should see the dashboard locally.
* Run `yarn ts-migration:enumerate`. You should see a success message. (Make sure to clear your working index as this changes a file that is kept in the repo.)

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
